File: 프로그래머스/3/12927. 야근 지수/야근 지수.py
import heapq
def solution(n, works):
    answer = 0
    
    q = []
    if n >= sum(works):
        return 0

    works = [-i for i in works]
    heapq.heapify(works)
    for _ in range(n):
        i = heapq.heappop(works)
        i +=1
        heapq.heappush(works, i)
    
    answer = sum([w**2 for w in works])
    
    return answer

File: 프로그래머스/3/12927. 야근 지수/야근 지수.java
import java.util.*;
class Solution {
    public long solution(int n, int[] works) {
        long answer = 0;
        PriorityQueue<Integer> pq = new PriorityQueue<>((a,b)->b-a);
        
        for(int w : works){
            pq.offer(w);
        }
        while(n>0 && !pq.isEmpty()){
            int x = pq.poll();
            x-=1;
            n-=1;
            if(x>0){
                pq.offer(x);
            }
            
        }
        if(pq.size()>0){
            for(int p:pq){
            answer += Math.pow(p,2);
            }
        }

        
        return answer;
    }
}